src/sagemaker_pytorch_serving_container/default_pytorch_inference_handler.py

```diff
@@ -101,8 +108,12 @@ def default_output_fn(self, prediction, accept):
         """
         if type(prediction) == torch.Tensor:
             prediction = prediction.detach().cpu().numpy().tolist()
-        encoded_prediction = encoder.encode(prediction, accept)
-        if accept == content_types.CSV:
-            encoded_prediction = encoded_prediction.encode("utf-8")
 
-        return encoded_prediction
+        for content_type in utils.parse_accept(accept):
+            if content_type in encoder.SUPPORTED_CONTENT_TYPES:
+                encoded_prediction = encoder.encode(prediction, content_type)
+                if content_type == content_types.CSV:
+                    encoded_prediction = encoded_prediction.encode("utf-8")
```

Comment:
I don't know if the `encode("utf-8")` call is actually necessary. If the prediction is encoded as JSON, then a string is returned. I figure that the server must automatically encode strings responses as UTF-8.

In any case, the `encode("utf-8")` was there before, so I've decided to keep it.